### PR TITLE
Make ONNX multithread safe

### PR DESCRIPTION
**Pull request by jeyserma**
Create a WeaverInterface object per thread.

Tested with many cores (1, 8, 32) and yields identical results.

### CODE REVIEW

**Comment by selvaggi** [comment]
many thanks, this is fixes inconsistency in inference results when MT vs single thread. I have also double-checked that results are identical, as shown in attachment. @kjvbrt @gganis @BrieucF  please merge as soon as possible. 

image

File: config/FCCAnalysisRun.py [hunk 1]
```diff
@@ -1058,12 +1058,15 @@ def run(mainparser, subparser=None):
...
+        print(args.command)
         args.command
         if args.command == "run":      runStages(args, rdfModule, args.preprocess, analysisFile)
         elif args.command == "final":  runFinal(rdfModule)
         elif args.command == "plots":  runPlots(analysisFile)
         return
-    except AttributeError:
+    except Exception as e:
+ 
+        print("Exception occurred for value '"+ elem + "': "+ repr(e))
```
Comment (kjvbrt) [inline config/FCCAnalysisRun.py:1069]: Can you provide an example how do you run the analysis script?

**Comment by kjvbrt** [comment]
Hello @jeyserma, @selvaggi,

running the first stage from the `examples/FCCee/weaver/README.md` crashes with
```
input_line_139:2:29: error: use of undeclared identifier 'magFieldBz'
auto lambda38 = [](){return magFieldBz[0]
                            ^
input_line_143:2:29: error: use of undeclared identifier 'magFieldBz'
auto lambda38 = [](){return magFieldBz[0]
                            ^
input_line_144:2:29: error: use of undeclared identifier 'magFieldBz'
auto lambda38 = [](){return magFieldBz[0]
                            ^
```
Full crash report: 
[crash.txt](https://github.com/HEP-FCC/FCCAnalyses/files/10842046/crash.txt)

File: config/FCCAnalysisRun.py (hunk 1 above)
Comment (jeyserma) [reply config/FCCAnalysisRun.py]: I removed this debugging line

**Comment by jeyserma** [comment]
To run the inference, an example command is:
`fccanalysis run examples/FCCee/weaver/analysis_inference.py --output inference_ss.root --files-list /eos/experiment/fcc/ee/generation/DelphesEvents/winter2023_training/IDEA/wzp6_ee_nunuH_Hss_ecm240/events_557817777.root --nevents 100`

I let @selvaggi comment on whether the stage1.py is necessary

**Comment by kjvbrt** [comment]
> To run the inference, an example command is: `fccanalysis run examples/FCCee/weaver/analysis_inference.py --output inference_ss.root --files-list /eos/experiment/fcc/ee/generation/DelphesEvents/winter2023_training/IDEA/wzp6_ee_nunuH_Hss_ecm240/events_557817777.root --nevents 100`
> 
> I let @selvaggi comment on whether the stage1.py is necessary

This crashes as well :)

```
09:36:49 [jsmiesko@carriacou FCCAnalyses (master=)]$ fccanalysis run examples/FCCee/weaver/analysis_inferen
ce.py --output inference_ss.root --files-list /eos/experiment/fcc/ee/generation/DelphesEvents/winter2023_tr
aining/IDEA/wzp6_ee_nunuH_Hss_ecm240/events_557817777.root --nevents 100
----> Load cxx analyzers from libFCCAnalyses... 
--------------loading analysis file   /home/jsmiesko/Work/FCC/onnx/FCCAnalyses/examples/FCCee/weaver/analys
is_inference.py
The variable <outputDir> is optional in your analysis.py file, return default value running dir
The variable <outputDirEos> is optional in your analysis.py file, return default empty string
----> Running with user defined list of files (either locally or from batch)
----> Create dataframe object from files: 
    /eos/experiment/fcc/ee/generation/DelphesEvents/winter2023_training/IDEA/wzp6_ee_nunuH_Hss_ecm240/event
s_557817777.root
----> nevents original=0  local=100
The variable <outputDir> is optional in your analysis.py file, return default value running dir
The variable <geometryFile> is optional in your analysis.py file, return default value empty string
The variable <readoutName> is optional in your analysis.py file, return default value empty string
----> Init done, about to run 100 events on 1 CPUs
Traceback (most recent call last):
  File "/cvmfs/sw-nightlies.hsf.org/spackages6/fccanalyses/commit.da3f53198ca408c0949d6924031dba40bbc0658e/
x86_64-centos7-gcc11.2.0-opt/6k47u/bin/fccanalysis", line 33, in <module>
    run(parser)
  File "/cvmfs/sw-nightlies.hsf.org/spackages6/fccanalyses/commit.da3f53198ca408c0949d6924031dba40bbc0658e/
x86_64-centos7-gcc11.2.0-opt/6k47u/python/config/FCCAnalysisRun.py", line 1062, in run
    if args.command == "run":      runStages(args, rdfModule, args.preprocess, analysisFile)
  File "/cvmfs/sw-nightlies.hsf.org/spackages6/fccanalyses/commit.da3f53198ca408c0949d6924031dba40bbc0658e/
x86_64-centos7-gcc11.2.0-opt/6k47u/python/config/FCCAnalysisRun.py", line 637, in runStages
    runLocal(rdfModule, args.files_list, args)
  File "/cvmfs/sw-nightlies.hsf.org/spackages6/fccanalyses/commit.da3f53198ca408c0949d6924031dba40bbc0658e/
x86_64-centos7-gcc11.2.0-opt/6k47u/python/config/FCCAnalysisRun.py", line 533, in runLocal
    runRDF(rdfModule, fileListRoot, outFile, nevents_local, args)
  File "/cvmfs/sw-nightlies.hsf.org/spackages6/fccanalyses/commit.da3f53198ca408c0949d6924031dba40bbc0658e/
x86_64-centos7-gcc11.2.0-opt/6k47u/python/config/FCCAnalysisRun.py", line 379, in runRDF
    df2 = getElement(rdfModule.RDFanalysis, "analysers")(df)
  File "/home/jsmiesko/Work/FCC/onnx/FCCAnalyses/examples/FCCee/weaver/analysis_inference.py", line 28, in 
analysers
    jetFlavourHelper = JetFlavourHelper(weaver_preproc, weaver_model, weaverConfig)
  File "/home/jsmiesko/Work/FCC/onnx/FCCAnalyses/addons/ONNXRuntime/python/jetFlavourHelper.py", line 47, i
n __init__
    weaver = JetFlavourUtils.setup_weaver(
TypeError: void FCCAnalyses::JetFlavourUtils::setup_weaver(const string&, const string&, const ROOT::VecOps
::RVec<string>&) =>
    TypeError: takes at most 3 arguments (4 given)
```

**Comment by jeyserma** [comment]
The code has been refactored by @selvaggi, as well as the unit tests.
I ran the unit tests on the PR and all went fine.

**Pull request merged**